### ISSUE TITLE
S4036 Searching OS commands in PATH is security-sensitive

### ISSUE THREAD

**Issue by eric-therond-sonarsource** [SonarSource/sonar-dotnet#3437]
Implements https://jira.sonarsource.com/browse/RSPEC-4036 (C# specific [RSPEC](https://jira.sonarsource.com/browse/RSPEC-5894) here).

Should raise when the cmd/file argument (first arg) of these functions:
_System.Diagnostics.ProcessStartInfo.ProcessStartInfo(string)
System.Diagnostics.ProcessStartInfo.ProcessStartInfo(string, string)
System.Diagnostics.Process.Start(string)
System.Diagnostics.Process.Start(string, string)
System.Diagnostics.Process.Start(string, string, System.Security.SecureString, string)
System.Diagnostics.Process.Start(string, string, string, System.Security.SecureString, string)
System.Diagnostics.ProcessStartInfo.FileName.set_

is not an absolute path or relative path (ie: the PATH environment variable is used to search the binary).

A check could be to look for the first chars of the string, if different from:
- ./
- .\
- ../
- ..\
- \
- /
- C:\

it should raise, example:
```
Process p = new Process();
p.StartInfo.FileName = "/usr/bin/binary"; // Compliant

Process p = new Process();
 p.StartInfo.FileName = "binary"; // Sensitive
```

Code examples: https://github.com/SonarSource/security-expected-issues/tree/master/dotnet/NetCore3MVC/Controllers/RSPEC_4036
